datafusion/src/optimizer/simplify_expressions.rs

```diff
@@ -0,0 +1,538 @@
+                _ => expr.clone(),
+            })
+            .unwrap_or_else(|| expr.clone()),
+        Expr::BinaryExpr {
+            left,
+            op: Operator::And,
+            right,
+        } if expr_contains(right, left) => as_binary_expr(right)
+            .map(|x| match x {
+                Expr::BinaryExpr {
+                    left: _,
+                    op: Operator::Or,
+                    right: _,
+                } => simplify(&*left.clone()),
+                Expr::BinaryExpr {
+                    left: _,
+                    op: Operator::And,
+                    right: _,
+                } => simplify(&x.clone()),
+                _ => expr.clone(),
+            })
+            .unwrap_or_else(|| expr.clone()),
+        Expr::BinaryExpr { left, op, right } => Expr::BinaryExpr {
+            left: Box::new(simplify(&left)),
+            op: *op,
+            right: Box::new(simplify(right)),
+        },
+        _ => expr.clone(),
+    }
+}
+
+fn optimize(plan: &LogicalPlan) -> Result<LogicalPlan> {
+    let new_inputs = plan
+        .inputs()
+        .iter()
+        .map(|input| optimize(input))
+        .collect::<Result<Vec<_>>>()?;
+    let expr = plan
+        .expressions()
+        .into_iter()
+        .map(|x| simplify(&x))
+        .collect::<Vec<_>>();
+    utils::from_plan(&plan, &expr, &new_inputs)
+}
+
+impl OptimizerRule for SimplifyExpressions {
+    fn name(&self) -> &str {
+        "simplify_expressions"
+    }
+
+    fn optimize(
+        &self,
+        plan: &LogicalPlan,
+        execution_props: &ExecutionProps,
+    ) -> Result<LogicalPlan> {
+        match plan {
+            LogicalPlan::Explain {
+                verbose,
+                plan,
+                stringified_plans,
+                schema,
+            } => {
+                let schema = schema.as_ref().to_owned().into();
+                optimize_explain(
+                    self,
+                    *verbose,
+                    &*plan,
+                    stringified_plans,
+                    &schema,
+                    execution_props,
+                )
+            }
+            _ => optimize(plan),
+        }
+    }
+}
+
+impl SimplifyExpressions {
+    #[allow(missing_docs)]
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::logical_plan::{and, binary_expr, col, lit, Expr, LogicalPlanBuilder};
+    use crate::test::*;
+
+    fn assert_optimized_plan_eq(plan: &LogicalPlan, expected: &str) {
+        let rule = SimplifyExpressions::new();
+        let optimized_plan = rule
+            .optimize(plan, &ExecutionProps::new())
+            .expect("failed to optimize plan");
+        let formatted_plan = format!("{:?}", optimized_plan);
+        assert_eq!(formatted_plan, expected);
+    }
+
+    #[test]
+    fn test_simplify_or_true() -> Result<()> {
+        let expr_a = binary_expr(col("c"), Operator::Or, lit(true));
```

Comment:
FWIW you can also create `or` operators using `Expr::or`

For example:

```suggestion
        let expr_a = col("c").or(lit(true)));
```